### PR TITLE
Demo_Syoten

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c [hunk 1]
```diff
@@ -28,45 +37,543 @@ const ActorInit Demo_Syoten_InitVars = {
...
+    this->unk_3E4 = 0;
+    this->unk_3E6 = 0;
+    this->unk_3DC = NULL;
+    this->unk_3E0 = NULL;
+    this->unk_3F2 = 0;
+    this->unk_3D8 = 1.0f;
+
+    switch (DEMOSYOTEN_GET_F(&this->actor)) {
+        case DEMOSYOTEN_F_0:
+            func_80183430((void*)this->unk_144, (void*)&object_syoten_Blob_001328, (void*)&object_syoten_Blob_00023C,
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c:76]: Are these void casts needed? It seems more likely that `unk_144` is a struct referenced by a pointer anyway.

File: src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c [hunk 2]
```diff
@@ -28,45 +37,543 @@ const ActorInit Demo_Syoten_InitVars = {
...
+        sp30.z = sp30.y;
+        sp24.z = sp24.y;
+        this->actor.shape.rot.z = -Math_Vec3f_Yaw(&sp24, &sp30);
+    }
+}
+
+typedef struct {
+    /* 0x00 */ char unk00[0xF];
+    /* 0x0F */ s8 unk_0F;
+} DemoSyotenStruct; // size = 0x10
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c:205]: Isn't this just a `Vtx`?

File: src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.h [hunk 3]
```diff
@@ -7,9 +7,41 @@ struct DemoSyoten;
...
+    /* 0x01 */ u8 count;
+    /* 0x02 */ UNK_TYPE1 unk2[0x16];
+} DemoSyotenStruct2; // almost certainly wrong, just filling space
+
 typedef struct DemoSyoten {
     /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x2B0];
+    /* 0x0144 */ UNK_TYPE1 unk_144[0x18];
+    /* 0x015C */ DemoSyotenStruct2* unk_15C;
+    /* 0x0160 */ UNK_TYPE1 unk160[0x14];
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.h:33]: This is probably all one struct, or at least a struct with substructs, given that almost none of the rest of it is referenced. But it's pretty meaningless without c_keyframe anyway.

File: src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.h [hunk 4]
```diff
@@ -7,9 +7,41 @@ struct DemoSyoten;
...
+    /* 0x0174 */ Vec3s unk_174[51];
+    /* 0x02A6 */ Vec3s unk_2A6[51];
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.h:35]: How did you determine the dimensions of these?

File: src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c (hunk 1 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c:76]: To suppress warnings they are yeah. UNK_TYPE is s32, this is a char[]. I know it's a struct but I have no idea what kind of struct and it seemed pointless to just make an empty/padded struct when I don't even know how big it is.

File: src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c (hunk 2 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c:205]: Maybe, no idea.

File: src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c (hunk 1 above)
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c:76]: Use `UNK_PTR` then.

File: src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.h (hunk 4 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.h:35]: Just guessed I think.

File: src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c (hunk 2 above)
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c:205]:
```c
DemoSyotenStruct* temp_v0 = Lib_SegmentedToVirtual(&object_syotenVtx_0018C0);
```
Given that this it's found by virtualising the pointer to a Vtx...

Comment (Kelebek1) [reply src/overlays/actors/ovl_Demo_Syoten/z_demo_syoten.c:205]: Oh, yeah, this was a thing I didn't see until I had done everything, and post-hoc fixed it at the end. I fix these syms at the end, and 0x18C0 wasn't in the object xml and so I started making a blob for it, but then thought I'd check the object anyway and then I saw it was in there as a Vtx, so I just replaced the symbol name and forgot the code. I'll change it ty ty.

File: include/variables.h [hunk 5]
```diff
@@ -3311,9 +3311,9 @@ extern ShrinkWindowContext gShrinkWindowContext;
 extern ShrinkWindowContext* gShrinkWindowContextPtr;
 // extern UNK_TYPE4 D_801F6B50;
 // extern UNK_TYPE1 D_801F6B58;
-extern FaultAddrConvClient sKaleidoAreaFaultClient;
 extern void (*sKaleidoScopeUpdateFunc)(GlobalContext* globalCtx);
 extern void (*sKaleidoScopeDrawFunc)(GlobalContext* globalCtx);
+extern FaultAddrConvClient sKaleidoAreaFaultClient;
```
Comment (hensldm) [inline include/variables.h:3316]: Looks like this bss is already imported in their correct `c` files. Is there any problems if you just comment them out?

File: include/z64animation.h [hunk 6]
```diff
@@ -341,4 +341,73 @@ typedef struct AnimationSpeedInfo {
     /* 0x0C */ f32 morphFrames;
 } AnimationSpeedInfo; // size = 0x10
 
+struct SkeletonInfo;
+
+typedef s32 (*UnkKeyframeCallback)(struct GlobalContext* globalCtx, struct SkeletonInfo* skeletonInfo, s32* arg2, Gfx** dList,
+                                   u8* arg4, void*);
```
Comment (hensldm) [inline include/z64animation.h:347]:
```suggestion
typedef s32 (*UnkKeyframeCallback)(struct GlobalContext* globalCtx, struct SkeletonInfo* skeletonInfo, s32* arg2, Gfx** dList,
                                   u8* arg4, void* arg5);
```

File: include/variables.h (hunk 5 above)
Comment (Kelebek1) [reply include/variables.h:3301]: Yeah it breaks bss all over.

**Review by EllipticEllipsis — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**